tests/test_tpa_pipeline.py

```diff
@@ -127,75 +127,37 @@ def test_get_consent_status(self):
             consent.delete()
             ec_user.delete()
 
-    def test_verify_data_sharing_consent(self):
+    def test_handle_enterprise_logistration(self):
         """
-        Test that we correctly verify consent status.
+        Test that we correctly handle incoming enterprise logistration requests.
         """
         backend = mock.MagicMock(name=None)
         with mock.patch('enterprise.tpa_pipeline.get_ec_for_running_pipeline') as fake_get_ec:
             fake_get_ec.return_value = None
-            assert verify_data_sharing_consent(backend, self.user) is None
+            assert handle_enterprise_logistration(backend, self.user) is None
         with mock.patch('enterprise.tpa_pipeline.get_ec_for_running_pipeline') as fake_get_ec:
-            fake_get_ec.return_value = mock.MagicMock(
-                requests_data_sharing_consent=False
-            )  # pylint: disable=redefined-variable-type
-            assert verify_data_sharing_consent(backend, self.user) is None
+            enterprise_customer = EnterpriseCustomerFactory(
+                enable_data_sharing_consent=False
+            )
+            fake_get_ec.return_value = enterprise_customer  # pylint: disable=redefined-variable-type
+            assert handle_enterprise_logistration(backend, self.user) is None
+            assert EnterpriseCustomerUser.objects.filter(
+                enterprise_customer=enterprise_customer,
+                user_id=self.user.id
+            ).count() == 1
         with mock.patch('enterprise.tpa_pipeline.get_ec_for_running_pipeline') as fake_get_ec:
             fake_get_ec.return_value = self.customer  # pylint: disable=redefined-variable-type
-            assert isinstance(verify_data_sharing_consent(backend, self.user), HttpResponseRedirect)
+            assert isinstance(handle_enterprise_logistration(backend, self.user), HttpResponseRedirect)
             ec_user = EnterpriseCustomerUser.objects.create(
```

Comment:
So, it turns out the Django test suite does this for free. So I've removed the `.delete()` calls. As for creation, the Factory method is about as good as we can get. :)